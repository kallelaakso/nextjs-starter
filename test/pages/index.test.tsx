import React from 'react';
import { act } from 'react-dom/test-utils';

import { render, fireEvent } from '../testUtils';
import IndexPage from '../../pages/index';

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<IndexPage />, {});
    expect(asFragment()).toMatchSnapshot();
  });

  it('clicking button triggers alert', () => {
    const { getByText } = render(<IndexPage />, {});
    window.alert = jest.fn();

    act(() => {
      fireEvent.click(getByText('Increase'));
    });
    expect(window.alert).toHaveBeenCalledWith('Increased');
  });
});
