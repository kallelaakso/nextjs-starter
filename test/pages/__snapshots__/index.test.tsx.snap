// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <h1
    class="pages__Title-sc-1q7pvn8-0 iqxFUP"
  >
    HOME
  </h1>
  <p>
    Counter: 1
  </p>
  <button>
    Increase
  </button>
  <button>
    Decrease
  </button>
</DocumentFragment>
`;
